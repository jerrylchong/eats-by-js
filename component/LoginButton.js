import React from "react";
import {Text, TouchableOpacity, StyleSheet, View} from "react-native";

const LoginButton = (prop) => {
    return (
        <View style = {styles.shadow}>
            <TouchableOpacity style = {styles.button} onPress = {prop.onPress}>
                <Text style = {{color: 'white'}}>{prop.text}</Text>
            </TouchableOpacity>
        </View>
    )
}

export default LoginButton

const styles = StyleSheet.create({
    shadow: {
        width: 200,
        height: 40,
        backgroundColor: '#ff6961',
        borderRadius: 20,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
        margin: 10
    },
    button: {
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    }
})