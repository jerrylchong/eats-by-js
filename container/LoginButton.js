import React from "react";
import {Text, TouchableOpacity, StyleSheet, View} from "react-native";

const LoginButton = (prop) => {
    return (
        <View style = {styles.shadow}>
            <TouchableOpacity style = {styles.button} onPress = {prop.onPress}>
                <Text>{prop.text}</Text>
            </TouchableOpacity>
        </View>
    )
}

export default LoginButton

const styles = StyleSheet.create({
    shadow: {
        width: 200,
        height: 40,
        backgroundColor: 'white',
        borderRadius: 20,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
        margin: 10
    },
    button: {
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    }
})