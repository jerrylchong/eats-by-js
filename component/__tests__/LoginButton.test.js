import React from 'react';
import renderer from 'react-test-renderer';
import LoginButton from "../LoginButton";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {TouchableOpacity} from "react-native";

test('renders correctly', () => {
    const tree = renderer.create(<LoginButton/>).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });
test('test onPress functionality', () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('onPress invoked');
    const wrapper = shallow(<LoginButton onPress={onPressEvent}/>);
    wrapper.find(TouchableOpacity).props().onPress();
    expect(onPressEvent.mock.calls.length).toBe(1);
})