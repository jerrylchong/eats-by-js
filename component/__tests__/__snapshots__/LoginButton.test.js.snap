// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ff6961",
        "borderRadius": 25,
        "elevation": 5,
        "height": 66.7,
        "margin": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
        "width": 333.5,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Ubuntu",
        }
      }
    />
  </View>
</View>
`;
